chain/pool/txlist.go

```diff
@@ -181,14 +202,35 @@ func (nst *NonceSortedTxs) Get(nonce uint64) (*transaction.Transaction, error) {
 }
 
 func (nst *NonceSortedTxs) GetAllTransactions() []*transaction.Transaction {
+	nst.mu.RLock()
+	defer nst.mu.RUnlock()
+
 	txns := make([]*transaction.Transaction, 0)
```

Comment:
We can change to `make([]*transaction.Transaction, nst.len())` for efficiency

Reply:
Done in additional commit.